Buildings/Fluid/CHPs/ThermalElectricalFollowing.mo

```diff
@@ -0,0 +1,371 @@
+    annotation (Placement(transformation(extent={{0,-140},{20,-120}})));
+  Buildings.Fluid.CHPs.BaseClasses.Controller opeMod(
+    final per=per,
+    final waitTime=waitTime) "Current operation mode"
+    annotation (Placement(transformation(extent={{40,150},{60,170}})));
+  Buildings.Controls.OBC.CDL.Logical.And runSig "Run if avaFlag and PEleDem non zero"
+    annotation (Placement(transformation(extent={{-20,120},{0,140}})));
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys(
+    final uLow=per.PEleMax/2*10^(-3),
+    final uHigh=per.PEleMax*10^(-3)) "Determine if demand larger than zero"
+    annotation (Placement(transformation(extent={{-60,100},{-40,120}})));
+  Buildings.Fluid.CHPs.BaseClasses.FilterPower fil(
+    final per=per) "Power after applied constrains"
+    annotation (Placement(transformation(extent={{-140,80},{-120,100}})));
+  Buildings.Fluid.CHPs.BaseClasses.WaterInternalControl conWat(
+    final per=per) if per.coolingWaterControl
+    "Internal controller for water flow rate"
+    annotation (Placement(transformation(extent={{120,130},{140,150}})));
+  Modelica.Blocks.Sources.RealExpression mWat_flow(
+    final y=port_a.m_flow) "Water flow rate"
+    annotation (Placement(transformation(extent={{-140,30},{-120,50}})));
+  Modelica.Blocks.Sources.RealExpression TWatIn(
+    final y=Medium.temperature(Medium.setState_phX(port_a.p, inStream(port_a.h_outflow))))
+    "Water inlet temperature"
+    annotation (Placement(transformation(extent={{-140,50},{-120,70}})));
+  Buildings.Fluid.CHPs.BaseClasses.AssertWatTem assWatTem(
+    final per=per)
+    "Assert if water temperature is outside boundaries"
+    annotation (Placement(transformation(extent={{120,-70},{140,-50}})));
+  Modelica.Thermal.HeatTransfer.Sensors.TemperatureSensor TWatOut
+    "Water outlet temperature"
+    annotation (Placement(transformation(extent={{60,-70},{80,-50}})));
+  Modelica.Thermal.HeatTransfer.Sensors.HeatFlowSensor watHea
+    "Heat transfer from the water control volume"
+    annotation (Placement(transformation(extent={{-10,10},{10,-10}},
+      rotation=-90, origin={-20,-100})));
+  Buildings.Fluid.CHPs.BaseClasses.PowerConsumption powCon(
+    final per=per) "Power consumption during stand-by and cool-down modes"
+    annotation (Placement(transformation(extent={{120,90},{140,110}})));
+  Modelica.Thermal.HeatTransfer.Sensors.HeatFlowSensor QLos
+    "Heat transfer to the surrounding"
+    annotation (Placement(transformation(extent={{-10,10},{10,-10}},
+      rotation=180, origin={-60,-140})));
+  Buildings.Controls.OBC.CDL.Continuous.LimPID cooWatCon(
+    final controllerType=watOutCon,
+    final k=k,
+    final Ti=Ti,
+    final Td=Td,
+    final yMax=yMax,
+    final yMin=yMin) if optionalFollowing
+    "Cooling water outplet controller"
+    annotation (Placement(transformation(extent={{-80,340},{-60,360}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterThreshold neeFolThe(
```

Comment:
didn't notice the `PIDHysteresis` model before. It would be good substitute.

Reply:
Addressed the PI reset with `theFol` signal. 

However (and after reconsidering) I do not understand the rationale behind the threshold of 1 degC on the water outlet temperature set point to reset PEle in case of thermal following.

Reply:
Addressed by suppressing the switch.